devicetypes/heltun/heltun-ft01-fan-coil-thermostat.src/heltun-ft01-fan-coil-thermostat.groovy

```diff
@@ -0,0 +1,571 @@
+		capability "Illuminance Measurement"
+		capability "Configuration"
+		capability "Health Check"
+		capability "Refresh"
+
+		attribute "thermostatFanState", "string"
+
+		fingerprint mfr: "0344", prod: "0004", model: "0002", deviceJoinName: "HELTUN Thermostat"
+	}
+	preferences {
+		input (
+			title: "HE-FT01 | HELTUN Fan Coil Thermostat",
+			description: "The user manual document with all technical information is available in support.heltun.com page. In case of technical questions please contact HELTUN Support Team at support@heltun.com",
+			type: "paragraph",
+			element: "paragraph"
+		)
+			parameterMap().each {
+			if (it.title != null) {
+			input (
+					title: "${it.title}",
+					description: it.description,
+					type: "paragraph",
+					element: "paragraph"
+				)
+			}
+			def unit = it.unit ? it.unit : ""
+			def defV = it.default as Integer
+			def defVDescr = it.options ? it.options.get(defV) : "${defV}${unit} - Default Value"
+			input (
+				name: it.name,
+				title: null,
+				description: "$defVDescr",
+				type: it.type,
+				options: it.options,
+				range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+				defaultValue: it.default,
+				required: false
+			)
+		}
+	}
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		return zwaveEvent(cmd)
+	}
+}
+
+def checkParam() {
+	boolean needConfig = false
+	parameterMap().each {
+		if (state."$it.name" == null || state."$it.name".state == "defNotConfigured") {
+			state."$it.name" = [value: it.default as Integer, state: "defNotConfigured"]
+			needConfig = true
+		}
+		if (settings."$it.name" != null && (state."$it.name".value != settings."$it.name" as Integer || state."$it.name".state == "notConfigured")) {
+			state."$it.name".value = settings."$it.name" as Integer
+			state."$it.name".state = "notConfigured"
+			needConfig = true
+		}
+	}
+	if ( needConfig ) {
+		configParam()
+	}
+}
+
+private configParam() {
+	def cmds = []
+	for (parameter in parameterMap()) {
+		if ( state."$parameter.name"?.value != null && state."$parameter.name"?.state in ["notConfigured", "defNotConfigured"] ) {
+				cmds << zwave.configurationV2.configurationSet(scaledConfigurationValue: state."$parameter.name".value, parameterNumber: parameter.paramNum, size: parameter.size).format()
+				cmds << zwave.configurationV2.configurationGet(parameterNumber: parameter.paramNum).format()
+				break
+		}
+	}
+	if (cmds) {
+		runIn(5, "checkParam")
+		sendHubCommand(cmds,500)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def parameter = parameterMap().find( {it.paramNum == cmd.parameterNumber } ).name
+	if (state."$parameter".value == cmd.scaledConfigurationValue){
+		state."$parameter".state = "configured"
+	}
+	else {
+		state."$parameter".state = "error"
+	}
+    log.info "Parameter ${parameter} Value ${state."$parameter".value} State ${state."$parameter".state}"
```

Comment:
formatting

Reply:
@dsobierajsk fixed!
I commit the changes by the Smartthings IDE but it seems like it is not committed.
Could you please let me know if there is another way ?